source/MaterialXCore/Types.h

```diff
@@ -173,6 +173,19 @@ template <class V, class S, size_t N> class VectorN : public VectorBase
         return *this;
     }
 
+    /// @}
+    /// @name Geometric operators
+    /// @{
+
+    /// Get the magnitude of a vector
+    S getMagnitude() const;
```

Comment:
This new method looks great, but I'm thinking that we may need to rename the existing "length" method to "numElements", since the original name is too easily confused for "return the magnitude of this vector".  What are your thoughts?